### PR TITLE
For Comment: Integrated Applications / Services experiment

### PR DESCRIPTION
**Pull request by jbarrett**
@nilnilnil @zekiel 

- Dancer2 for easy Plack routing and plugin ecosystem
- PSGI Session used by Catalyst and other apps
- Existing XSlate member of DDGC used to render
  - Shim functions for essential (i.e. prevent crashing) catalyst context, view functionality
  - Some stuff not working, like `chained_uri`
- Serializer used for "API" service

Pretty crusty, but it's a jumping off point for discussion I suppose.

"App" Screenie - avatars, links not working since they rely on chained_uri:

Links not working

### CODE REVIEW

File: lib/DDGC/Web/App/UserLoggedIn.pm [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    },
+};
+
+set session => 'PSGI';
+
+get '/' => sub {
+    my $user = schema->resultset('User')->find({ username => session('__user') });
+    return 'None' if !$user;
+    my @templates = ( qw/base.tx loggedinuser.tx/ );
+    $xslate->{functions}->{next_template} = sub {
```
Comment (jbarrett) [inline lib/DDGC/Web/App/UserLoggedIn.pm]: This shouldn't be here, it's a code ref passed in vars, see line 38

**Comment by zekiel** [comment]
cool! Also ready to chat about it Monday or marinate first?

**Comment by jbarrett** [comment]
@zekiel Nope, any time you want is good - this gave me a picture of some next steps.

**Comment by zekiel** [comment]
sent an invite for 11 on Tue. Only bc my schedule for Mon is sad :(

**Comment by jbarrett** [comment]
Cool, I'll probably have experiment two by then. Maybe three :)

**Comment by nilnilnil** [comment]
Are you all set here @jbarrett?

**Comment by jbarrett** [comment]
Thanks @nilnilnil, something more complete and consistent coming up!

**Pull request closed without merging**